### ISSUE TITLE
Memory alerts feature?

### ISSUE THREAD

**Issue by nojhan** [liquidprompt/liquidprompt#52]
Should we have a memory meter (just like the battery and load ones)?

Emmanuel Rouat have suggested (some weeks ago) to also add a swap detector:

```
# Experimental: detect if machine is swapping
function swap_color()
{
    if [ $(vmstat | awk 'END{ print $7 }') -gt "0" ]; then  # swapin > 0
        echo -en ${ALERT}
    fi
}
```

**Comment by Rycieos** [comment]
I came up with an implementation for Linux. It doesn't use any forks or calls, and only loads one file, `/proc/meminfo`.

```bash
mempresure() {  # return: percentage of memory free
  local id value total available
  while IFS=$' \t' read -r id value _ ; do
    if [[ $id == 'MemTotal:' ]]; then
      total=$value
    elif [[ $id == 'MemAvailable:' ]]; then
      available=$value
    fi
  done < /proc/meminfo
  printf '%s\n' "$(( available * 100 / total ))"
}
```

I just don't know how useful this would be. Is there still anyone who would actually want this?

**Comment by nojhan** [comment]
Having myself regular memory problems, I would be interested. This may be disabled by default, of course.

**Comment by Rycieos** [comment]
Merged in #770
